### PR TITLE
support vscode variable in debugConfig environment

### PR DESCRIPTION
**Pull request by upcliujie**
### This changes src\cmakeProject.ts

The following changes are proposed:
``` TypeScript
if (debugConfig.environment) {
    for (const env of debugConfig.environment) {
        env.value = await expandString(env.value, await this.getExpansionOptions());
    }
}
```

## The purpose of this change
resolve the vscode variable in debugConfig environment

### CODE REVIEW

**Comment by gcampbell-msft** [comment]
@upcliujie These changes look great! However, before merging, make a CHANGELOG.md update to keep it up to date. Please feel free to give yourself credit following the pattern already present. Thanks!

**Comment by bobbrow** [comment]
If you're adding support for a new property in `cmake.debugConfiguration` it is recommended to update the schema for the setting in `package.json` so that customers will get IntelliSense for it.

**Comment by upcliujie** [comment]
> @upcliujie These changes look great! However, before merging, make a CHANGELOG.md update to keep it up to date. Please feel free to give yourself credit following the pattern already present. Thanks.

Changelog updated

**Comment by upcliujie** [comment]
@microsoft-github-policy-service agree [company="{kylinsoft}"]

**Comment by upcliujie** [comment]
@microsoft-github-policy-service agree company="kylinsoft"

**Comment by gcampbell-msft** [comment]
@upcliujie Please update the schema for debugConfig as well, so that customers will get IntelliSense for this. THanks!

**Comment by upcliujie** [comment]
> @upcliujie Please update the schema for debugConfig as well, so that customers will get IntelliSense for this. THanks!

Sorry, I  don't know how to update the schema for debugConfig. Can you give me some help?

**Comment by upcliujie** [comment]
> @upcliujie Please update the schema for debugConfig as well, so that customers will get IntelliSense for this. THanks!

Sorry, I  don't know how to update the schema for debugConfig. Can you give me some help?

**Comment by gcampbell-msft** [comment]
> > @upcliujie Please update the schema for debugConfig as well, so that customers will get IntelliSense for this. THanks!
> 
> @gcampbell-msft Sorry, I don't know how to update the schema for debugConfig. Can you give me some help?

Actually, looking at this again, no new property is supported, it's just adding variable expansion to the current properties. Approved.

**Review by gcampbell-msft — APPROVED** [review]

**Comment by bobbrow** [comment]
> > > @upcliujie Please update the schema for debugConfig as well, so that customers will get IntelliSense for this. THanks!
> > 
> > 
> > @gcampbell-msft Sorry, I don't know how to update the schema for debugConfig. Can you give me some help?
> 
> Actually, looking at this again, no new property is supported, it's just adding variable expansion to the current properties. Approved.

Sorry, I must not have looked at the code.  `debugConfig.environment` already exists in the schema.

File: CHANGELOG.md
Comment (bobbrow) [inline CHANGELOG.md:36]: Suggestion:  "Add support for variable expansion in \`debugConfig.environment\`"

File: src/cmakeProject.ts
Comment (bobbrow) [inline src/cmakeProject.ts:2665]: I recommend calling this once outside the loop and saving the value to reuse here.

File: CHANGELOG.md
Comment (upcliujie) [reply CHANGELOG.md:36]: OK

**Comment by upcliujie** [comment]
@gcampbell-msft @gcampbell-msft  Code updated

**Review by gcampbell-msft — APPROVED** [review]

**Review by bobbrow — DISMISSED** [review]
👍

**Comment by gcampbell-msft** [comment]
@upcliujie Could you please fix the linter error?

**Comment by upcliujie** [comment]
@gcampbell-msft linter error has been fixed.

**Review by gcampbell-msft — APPROVED** [review]

**Comment by gcampbell-msft** [comment]
This has already passed the tests and gotten approval, no need to run the tests again. Merging.

**Pull request merged**
